### PR TITLE
fix: resolve refs in EnumArrayRenderer

### LINKED ISSUE #2192: EnumArray Renderer not working with $refs
### Describe the bug

The EnumArray renderer works without $refs in the schema, but with $refs, those EnumOptions do not seem to get handled. 

The 'options' prop should have returned the list of enum options, but comes back undefined. There seems to be no other prop to use other than 'options'  in ControlProps, OwnPropsOfEnum, DispatchPropsOfMultiEnumControl.

Here is the sample schema:
 schemaWithRef = {
      $schema: 'http://json-schema.org/draft-07/schema#',
      additionalProperties: false,
      type: 'object',
      definitions: {
        Partner: {
          enum: ['DeutscheTelekom', 'GoDaddy', 'Liquid', 'Rogers'],
          type: 'string'
        }
      },
      properties: {
        partner: {
          items: {
            $ref: '#/definitions/Partner'
          },
          type: 'array'
        }
      }
    }

### Expected behavior

The expected behavior is that the 'options' prop should return ['DeutscheTelekom', 'GoDaddy', 'Liquid', 'Rogers'], but instead returns undefined.

### Steps to reproduce the issue

Dispatch JSON forms with the given sample schema.

### Screenshots

_No response_

### In which browser are you experiencing the issue?

Edge 118

### Which Version of JSON Forms are you using?

v3.1.0

### Framework

React

### RendererSet

_No response_

### Additional context

FluentUI

### PR DESCRIPTION
**Pull request by LukasBoll**
resolve refs ind the MultiEnumControlProps mapper and tester

fixes #2192

### CODE REVIEW

**Review by lucas-koehler — CHANGES_REQUESTED** [review]
Hi @LukasBoll the changes look pretty good to me. There is one typo fix inline that also fixes the out of memory of the tests. I'm not sure why the typo led to an out of memory but it should not be related to this PR:
Likely it is ue to the implementation of `mapStateToMultiEnumControlProps`. However, I did not see anything obvious there. Furthermore, in a running application the method should never be invoked because the tester does not apply for effectively empty `items`. I tested this in the example material app and there simply a `No applicable renderer found.` is shown.

File: packages/core/test/util/renderer.test.ts [hunk 1]
```diff
@@ -983,6 +983,50 @@ test('mapStateToMultiEnumControlProps - enum items', (t) => {
...
+              type: 'string',
+              enum: ['red', 'green', 'pink'],
+            },
+          },
+          type: 'object',
+          properties: {
+            colors: {
+              type: 'array',
+              items: {
+                ref$: '#/definitions/colors',
```
[the remark this answers is not in the document]
Comment (LukasBoll) [reply packages/core/test/util/renderer.test.ts]: Oh, what a stupid mistake. The test is working now, thank you for taking a look @lucas-koehler !
I think the out of memory issues occurred in a view other test cases as well. 
So maybe there is an issue with one of the core functionalities, that is called by `mapStateToMultiEnumControlProps`.

[the remark this answers is not in the document]
Comment (lucas-koehler) [reply packages/core/test/util/renderer.test.ts]: Yes, I am also aware of at least on other PR with out of memory issues. Thanks for the hint. However, we don't need to look at that in this PR :D

**Review by lucas-koehler — APPROVED** [review]
LGTM now! Thanks for the contribution :)

**Pull request merged**